atomic_reactor/plugins/pre_add_help.py

```diff
@@ -0,0 +1,86 @@
+"""
+Copyright (c) 2015 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+
+
+Convert a help markdown file a man page and store it to /help.1 in the image
+so that 'atomic help' could display it.
+This is accomplished by appending an ADD command to it.
+
+Example configuration:
+{
+    'name': 'add_help',
+    'args': {'help_file': 'help.md'}
+}
+"""
+
+import os
+from subprocess import check_output, CalledProcessError, STDOUT
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.util import df_parser
+
+
+class AddHelpPlugin(PreBuildPlugin):
+    key = "add_help"
+    man_filename = "help.1"
+
+    def __init__(self, tasker, workflow, help_file="help.md"):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param help_file: filename of the markdown help file
+        """
+        # call parent constructor
+        super(AddHelpPlugin, self).__init__(tasker, workflow)
+        self.help_file = help_file
+
+    def run(self):
+        """
+        run the plugin
+        """
+
+        help_path = os.path.join(self.workflow.builder.df_dir, self.help_file)
+
+        if not os.path.exists(help_path):
+            self.log.info("File %s not found", help_path)
+            return
+
+        try:
+            check_output(["go-md2man", "-h"], stderr=STDOUT)
+        except OSError:
+            raise RuntimeError(
+                "Help file is available, but go-md2man is not present in a buildroot")
+        except CalledProcessError as ex:
+            # Ignore exit code 2, because this is what go-md2man does
+            if ex.returncode != 2:
+                raise RuntimeError(
+                    "There was an error with go-md2man, help returned %s" % ex.returncode)
+
+        man_path = os.path.join(self.workflow.builder.df_dir, self.man_filename)
+        cmd = "go-md2man -in=%s -out=%s" % (help_path, man_path)
```

Comment:
Security-aware programs should use fixed strings when passing them to the shell, so they know exactly how quoting will work.